src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
     public void add(T value) {
-
+        if (size >= elementData.length) {
+            elementData = increaseSizeArray(elementData);
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size >= elementData.length) {
+            elementData = increaseSizeArray(elementData);
+        }
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Yuo can't add "
+                    + "element by this index");
+        }
+        System.arraycopy(elementData, index, elementData,
+                index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("You can't use this "
+                    + "index");
+        }
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("You can't set value "
+                    + "to this position");
+        } else {
+            elementData[index] = value;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T temp = null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Element by this "
+                    + "index doesn't exist");
+        } else {
+            temp = elementData[index];
+            elementData[index] = null;
+            moveElementsAfterRemove(index);
+            size--;
+        }
+        return temp;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        if (element == null) {
+            for (int i = 0; i < elementData.length; i++) {
+                if (elementData[i] == null) {
+                    moveElementsAfterRemove(i);
+                    size--;
+                    return null;
+                }
+            }
+        }
+        for (int i = 0; i < elementData.length; i++) {
+            if (elementData[i] != null && elementData[i].equals(element)) {
+                moveElementsAfterRemove(i);
+                size--;
+                return element;
+            }
+        }
+        try {
+            throw new NoSuchFieldException();
+        } catch (NoSuchFieldException e) {
+            throw new NoSuchElementException();
+        }
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private T[] increaseSizeArray(T[] elementData) {
```

Comment:
```suggestion
    private void increaseSizeArray(T[] elementData) {
```

I think we could make this method void